CAIPs/caip-tbd.md

```diff
@@ -0,0 +1,102 @@
+
+
+## Backwards Compatibility
+
+Wallet extensions SHOULD consider adding a "developer mode" toggle via a UX so that dApp developers have the capability to disable the insecure context (http) check for the http://localhost:<any-port> origin only in the event that [localhost][w3c-secure-context-localhost] does not return `true` for secure context</a>. This will allow dApp developers to be able to continue to host dApps on the localhost origin if a browser environment has chosen to not already consider localhost a secure context. Most major browser providers do consider localhost a secure context already. This toggle MUST be set to disabled by default.
+
+## Test Cases
+
+### Required Test Cases
+
+- Top level `http://a.com` -> blocked (insecure/top level)
+- Top level `https://a.com` -> allowed
+- Top level `https://a.com` with `<iframe src="http://a.com/">` -> blocked (insecure first party iframe)
+- Top level `http://a.com` with `<iframe src="https://a.com/">` -> blocked (insecure top level window)
+- Top level `https://a.com` with `<iframe src="https://a.com">` -> allowed
+- Top level `https://a.com` with `<iframe src="https://b.com">` -> blocked (3p iframe without sufficient privileges)
+- Top level `https://b.com` with `<iframe src="http://a.com/">` with `<iframe src="https://b.com">` -> blocked (insecure iframe)
+- Top level `https://b.com` with `<iframe src="https://a.com">` with `<iframe src="https://b.com">` -> blocked (3p iframe without sufficient privileges)
+- Top level `https://a.com` with `<iframe src="https://sub.a.com">` -> allowed
+- Top level `https://a.com` with `<iframe src="https://a.com" sandbox>` -> blocked (sandbox attribute without "allow-same-origin")
+- Top level `https://a.com` with `<iframe src="https://a.com" sandbox="allow-same-origin allow-scripts">` -> allowed (but note this case is discouraged in https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#attr-sandbox because it’d allow the iframe to remove its own sandbox attribute)
+- Top level `data://foo with <iframe src="data://bar">` -> blocked (insecure top level scheme)
+- Top level `file://foo with <iframe src="file://bar">` -> blocked (3p iframe)
+- Top level `https://a.com` with `<iframe src="https://b.com" sandbox="allow-same-origin allow-scripts">` -> blocked (3p iframe without sufficient privileges)
+
+### Optional Test Cases
+
+- Top level `https://a.com with <iframe src="https://b.com" allow="ethereum">` -> Ethereum allowed, Solana blocked (3P iframe because "allow" attribute in use for ethereum)
+- Top level `https://a.com with <iframe src="https://b.com" allow="solana">` -> Solana allowed, Ethereum blocked (3P iframe because "allow" attribute in use for solana)
+- Top level `https://a.com with <iframe src="https://b.com" allow="ethereum solana">` -> Solana allowed, Ethereum allowed (3P iframe because "allow" attribute in use for solana and ethereum)
```

Comment:
I'm wondering if it makes sense to make this text more chain agnostic here? E.g. "web3" might make more sense here, but as discussed in https://github.com/w3c/webextensions/issues/271 we may run the risk of extensions colliding on registering this permission in a way that could produce an odd behavior. We'll need to consider the tradeoffs here if we're going to aim for a non unique `allow` attribute.

Reply:
+1 to making it more chain agnostic. 

OR lean into it being chain specific and use the chain ids themselves?

Reply:
Yeah that's the part I'm not certain about right now. Brave has taken the chain specific approach right now, but I'd be interested to see what others take is on this. I venture to guess that from a dApps perspective in a truly multichain world this would be a hindrance, but today it would be a nuance of just building something.

Reply:
not sure i understand what you mean by "chain agnostic"-- here at CASA we generally distinguish between "namespaces" (like `ethereum` or `solana`) which are VMs+RPC featuresets and "chains" (like ethereum main-net, i.e. `eip155:1`, or solana devnet, i.e. `solana: 8E9rvCKLFQia2Y35HXjjpWzj8weVo44K`) which are specific networks within them.

i guess i'm confused about the namespace of the allow string. the two options are currently `ethereum` and `solana`, because those are the two `window.x` options in today's web3js.  those map to CASA namespaces, perhaps sheerly by dumb luck-- what happens when `polkadot` and `cosmos` show up and their wallets start jockeying for "first permissions" on that namespace? things get complicated when someone wants to use `statemint` and `statemine` or `kusama` as the scope of allowance...

unless you're offering for the CASA namespaces repo to be the definitely registry of namespacing, in which case I'm all for it! 😉

Reply:
If this risks expanding into a relatively wide namespace, it might be a good idea to consider prefixing names (eg. `web3-ethereum`). My concern is the potential for conflict with other `allow` values. Feature policies already has about 30 keywords and new ones are added regularly.

Reply:
> If this risks expanding into a relatively wide namespace

My hope would be that it doesn't in order to produce better web compatibility between the different wallets, but I'm not certain how to handle the conflict in a way that wouldn't introduce security issues or conflicts in terms of the wallet that's being injected. E.g. it wouldn't be good if the user has two extension based wallets and wallet A injected (via content scripts) first into the main window and wallet B injected first into the iframe.

Reply:
I just got a chance to talk with others on our team familiar with the feature and it seems like we'd be good to drop support of this from this spec and if we're able to find a way to get this working with extensions in the future we can extend this spec like @coryhardman suggested or bring it back in if we've not reached final call for it yet. I'll make the updates here and in the EIP